<launch>
    <!-- Start the low-level driver manager with parameters -->
    <node name="ttyUSB0_manager" pkg="ax12_controller_core" type="controller_manager.py" required="true" output="screen">
        <param name="port_name" type="str" value="/dev/ttyUSB0"/>
        <param name="baud_rate" type="int" value="1000000"/>
        <param name="max_motor_id" type="int" value="25"/>
        <param name="update_rate" type="int" value="25"/>
    </node>
    
    <!-- Load joint controller configuration from YAML file to parameter server -->
    <rosparam file="$(find ax12_controller_core)/test/action_trajectory_controllers.yaml" command="load"/>
    <node name="upper_controller_spawner" pkg="ax12_controller_core" type="controller_spawner.py"
          args="--port /dev/ttyUSB0
                upper_controller"
          output="screen" >
    </node>
</launch>
